Sarah-Code-Review

sarahschaefer1 opened the pull request.

sarahschaefer1 commented on xmlproject/xmlproject/Pages/AboutYumSafe.cshtml:
```diff
@@ -38,14 +38,16 @@
 </style>
 
 <body style="background-color:#d5dce8;">
+	<strong>
```
Changed the text on the "about us" page to bold font. The Papyrus font was thin with inconsistent coloring before making it hard to read.

discospiff replied:
I urge caution here.  If everything is strong (bold), then nothing is strong.  If you want to adjust font, the best practice is to use CSS.

sarahschaefer1 commented on xmlproject/xmlproject/Pages/AboutYumSafe.cshtml:
```diff
@@ -38,14 +38,16 @@
+	<strong>
 	<h2> Welcome to YumSafe:Eat Assured</h2>
 	<p>At YumSafe, we believe that everyone deserves a delightful dining experience, free from worries about food safety and hygiene. We understand the importance of enjoying a meal at a restaurant where food quality is paramount and where health and safety standards are met without compromise. That’s why we created the YumSafe app – your ultimate companion for finding the best restaurants without v ...
-	<strong>What Sets Us Apart? </strong>
```
deleted the <strong> tags on the individual lines since I changed all of the body's text to "strong"

discospiff replied:
See above.

sarahschaefer1 commented on xmlproject/xmlproject/Pages/AboutYumSafe.cshtml:
```diff
@@ -39,15 +39,15 @@
 <body style="background-color:#d5dce8;">
 	<strong>
-	<h2> Welcome to YumSafe:Eat Assured</h2>
+	<h1> Welcome to YumSafe:Eat Assured</h1>
```
Enlarged the title more to make it stand out

discospiff replied:
I agree with the outcome, but not the justification; and both for the same reason.  h tags should not be used for formatting text.  They should be used for categorizing text.  Use CSS to format.  So, in this case, yes, the first h tag should be an h1 (as you wouldn't start a sentence without a paragraph)... but it has more to do with categorization than it does with look and feel.

sarahschaefer1 commented on xmlproject/xmlproject/Pages/AboutYumSafe.cshtml:
```diff
@@ -39,15 +39,15 @@
+	<h1> Welcome to YumSafe:Eat Assured</h1>
 	<p>At YumSafe, we believe that everyone deserves a delightful dining experience, free from worries about food safety and hygiene. We understand the importance of enjoying a meal at a restaurant where food quality is paramount and where health and safety standards are met without compromise. That’s why we created the YumSafe app – your ultimate companion for finding the best restaurants without v ...
-	What Sets Us Apart?
+	<h2>What Sets Us Apart?</h2>
```
Enlarged the headings of each section to make them stand out more

discospiff commented on xmlproject/xmlproject/Pages/AboutYumSafe.cshtml:
```diff
@@ -38,14 +38,16 @@
-	<strong>Our Commitment to You</strong>
+	<h2>Our Commitment to You</h2>
 	<p>At YumSafe, we are committed to promoting a culture of food safety and transparency in the restaurant industry. We believe that by encouraging restaurants to maintain high standards of hygiene, we contribute to the well-being of our community. Our platform acts as a bridge between conscientious diners and responsible restaurateurs, fostering an environment where everyone benefits.</p>
-	<strong>Join the YumSafe Community </strong>
+	<h2>Join the YumSafe Community</h2>
```
Ditto above.  I agree with the outcome, but not the justification.

discospiff commented on xmlproject/xmlproject/Pages/Index.cshtml:
```diff
@@ -245,6 +245,7 @@
     </div>
     <h2></h2>
 
+    <div class="footer" style="font-family: 'Times New Roman', Times, serif;">
```
It looks like you've moved an inline style to an embedded stylesheet.  While this is a good concept, there is a difference in selectors.  Here, it applies to everything in the div.  In the selector, it's only p tags within div tags.

discospiff commented on xmlproject/xmlproject/Pages/ReviewForm.cshtml:
```diff
@@ -2,17 +2,6 @@
 }
-<html>
-<head>
-    <meta charset="utf-8">
-    <meta name="viewport" content="width=device-width, initial-scale=1">
-    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
-    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.1/jquery.min.js"></script>
-    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js"></script>
-    <link rel="stylesheet" href="~/css/site.css" asp-append-version="true" />
```
These tags do not belong here, but... were the links, metas, and scripts moved to _Layout.chstml?  I don't see changes to that file in this commit.  If it was not, then functionality has been removed.

discospiff commented:
Good discussion here, but it needs a few more steps before I'd merge it.  Thus, I don't recommend merging this branch.

sarahschaefer1 commented:
Write up: 

Change 1- Changed the text on the "about us" page to bold font. A comment was made that "if all of the text is bolded, then none of it was bolded." My argument is that before, the Papyrus font was a good shape that fit the restaurant theme, but it was a thin and "broken" font, making it hard to read. I played with inserting other fonts, but the most aesthetically pleasing was keeping the font and making it bold. This eliminated the "brokenness" of the font making it composed of clean black lines that were much easier to read.
Change 2- Changed the headings of each section on the "about us" page to stand out from the body by enlarging the titles of each one (before they were the same size as the body) and making the title even larger to stand out from the other headings. Improved the look and readability.
Change 3- Standardized the font in the footer section of the home page. Before, there were two fonts being mixed that created an unprofessional look. I changed all of the text in the footer to the Times New Roman font to suit the rest of the page and for all of the text to match.
Change 4- Pulled the titles of the three columns of the footer to align at the top of the footer. Before, all three footer columns were scattered at different horizontal points making it harder to read and tainted the professionalism of the home page. I changed the line "align items: flex-end" to "align items: flex-start" for everything to become aligned horizontally in the footer.

- Three things I learned from completing the code review were: how to collaborate on a project with many others using GitHub and separate branches integrated with visual studio, how to have a high attention to detail to be able to read and understand one another's code and come up with unique changes, and how small changes in code can make large differences in the visual appeal and readability of a website.

- Yes, the project was available in GitHub on time, compiled correctly, and had adequate documentation.

sarahschaefer1 commented on xmlproject/xmlproject/Pages/AboutYumSafe.cshtml:
Bolded Font on "About Us" Page:
    * Reducing Technical Debt: Enhanced user experience and potentially reducing support requests related to text legibility issues	
Enlarged Headings on "About Us" Page:
    * Reducing Technical Debt: By standardizing and improving the visual hierarchy of the headings, I enhanced the readability and maintainability of the code. Clearer visual structure may reduce the chance of errors in future updates, and the <h> tags make it easier for developers to understand and modify the code.
Standardized Font in the Footer Section:
    * Reducing Technical Debt: Consistency in font usage across the website contributes to maintainability. It simplifies the style sheets, making it easier for developers to understand and modify styles in the future. It also reduces the risk of conflicts and errors that might arise from using multiple fonts.
Aligned Titles in Footer Columns:
    * Reducing Technical Debt: Aligning the titles in the footer columns improves the visual consistency and makes the code more maintainable. By using a consistent alignment approach, you reduce the chances of layout-related bugs.

Source cited: https://www.techtarget.com/searchcio/feature/The-negative-impact-of-technical-debt
Highlights 12 negative impacts of technical debt, each with some of the causes. I used this information to make changes to avoid these common pitfalls leading to technical debt
